app/src/main/kotlin/compiler/common/dfa/state_dfa/DfaState.kt

```diff
@@ -0,0 +1,6 @@
+package compiler.common.dfa.state_dfa
+
+interface DfaState<A, R> {
+    val result: R?
+    fun getPossibleSteps(): Map<A, DfaState<A, R>>
```

Comment:
Seems somewhat inconsistent that `result` is a property and `getPossibleSteps` is a method. (Maybe we could also replace any `getX` with `x` elsewhere in this project.)

Reply:
Agree, changed most getters to properties, at least in `parser` module